xbmc/application/Application.cpp

```diff
@@ -878,7 +878,16 @@ void CApplication::Render()
     return;
 
   // render gui layer
-  if (appPower->GetRenderGUI() && !m_skipGuiRender)
+  const bool renderGUI = appPower->GetRenderGUI();
+  if (m_guiRenderLastState != std::nullopt && renderGUI && m_guiRenderLastState != renderGUI)
+  {
+    CGUIComponent* gui = CServiceBroker::GetGUI();
+    if (gui)
+      CServiceBroker::GetGUI()->GetWindowManager().MarkDirty();
+  }
+  m_guiRenderLastState = renderGUI;
```

Comment:
I think you don't need this here. GetRenderGUI returns a simple bool, so the code can be simplified heavily. Might be I overlook some inheritance somewhere, but don't see it.

So the logic here is: whenever GetRenderGUI yields something that is not the current state it is marked dirty. Are you sure that this not simply disables dirtyregion entirely?

Normally I would assume - that we render when something is dirty. In this code though, we makr dirty whenever internal state and GetRenderGUI disagree.

Reply:
Went with the suggestion from @sarbes for now so this doesn't apply anymore.